src/go/k8s/controllers/redpanda/console_controller.go

```diff
@@ -71,9 +74,13 @@ func (r *ConsoleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 
 	var cluster redpandav1alpha1.Cluster
 	if err := r.Get(ctx, types.NamespacedName{
-		Namespace: console.Spec.ClusterKeyRef.Namespace,
-		Name:      console.Spec.ClusterKeyRef.Name,
+		Namespace: console.GetNamespace(),
+		Name:      console.GetName(),
```

Comment:
We shouldn't assume that. We should be flexible.

Reply:
Updated to index fields and reconcile cluster from `clusterKeyRef`